doc/plotting.rst

```diff
@@ -0,0 +1,313 @@
+each of the axes should be.
+
+.. ipython:: python
+
+    @savefig plotting_example_2d_simple.png width=4in
+    a.plot()
+
+It may seem strange that
+the the values on the y axis are decreasing with -0.5 on the top. This is because
+the pixels are centered over their coordinates, and the
+axis labels and ranges correspond to the values of the
+coordinates. 
+
+All 2d plots in xray allow the use of the keyword arguments ``yincrease=True``
+to produce a
+more conventional plot where the coordinates increase in the y axis.
+``xincrease`` works similarly.
+
+.. ipython:: python
+
+    @savefig 2d_simple_yincrease.png width=4in
+    a.plot(yincrease=True)
+
+Simulated Data
+~~~~~~~~~~~~~~
+
+For further examples we generate two dimensional data by computing the Euclidean
+distance from a 2d grid point to the origin.
+
+.. ipython:: python
+
+    x = np.arange(start=0, stop=10, step=2)
+    y = np.arange(start=9, stop=-7, step=-3)
+    xy = np.dstack(np.meshgrid(x, y))
+
+    distance = np.linalg.norm(xy, axis=2)
+
+    distance = xray.DataArray(distance, zip(('y', 'x'), (y, x)))
+    distance
+
+Note the coordinate ``y`` here is decreasing.
+This makes the y axes appear in the conventional way.
+
+.. ipython:: python
+
+    @savefig plotting_2d_simulated.png width=4in
+    distance.plot()
+
+Changing Axes
+~~~~~~~~~~~~~
+
+To swap the variables plotted on vertical and horizontal axes one can
+transpose the array.
+
+.. ipython:: python
+
+    @savefig plotting_changing_axes.png width=4in
+    distance.T.plot()
+
+To make x and y increase:
+
+.. ipython:: python
+
+    @savefig plotting_changing_axes2.png width=4in
+    distance.T.plot(xincrease=True, yincrease=True)
+
+Nonuniform Coordinates
+~~~~~~~~~~~~~~~~~~~~~~
+
+It's not necessary for the coordinates to be evenly spaced. If not, then
+:py:meth:`xray.DataArray.plot` produces a filled contour plot by calling
+:py:meth:`xray.DataArray.plot_contourf`. This example demonstrates that by
+using one coordinate with logarithmic spacing.
+
+.. ipython:: python
+
+    x = np.linspace(0, 500)
+    y = np.logspace(0, 3)
+    xy = np.dstack(np.meshgrid(x, y))
+    d_ylog = np.linalg.norm(xy, axis=2)
+    d_ylog = xray.DataArray(d_ylog, zip(('y', 'x'), (y, x)))
+
+    @savefig plotting_nonuniform_coords.png width=4in
+    d_ylog.plot()
+
+Calling Matplotlib
+~~~~~~~~~~~~~~~~~~
+
+Since this is a thin wrapper around matplotlib, all the functionality of
+matplotlib is available. For example, use a different color map and add a title.
+
+.. ipython:: python
+
+    d_ylog.plot(cmap=plt.cm.Blues)
+    plt.title('Euclidean distance from point to origin')
+
+    @savefig plotting_2d_call_matplotlib.png width=4in
+    plt.show()
+
+Colormaps
+~~~~~~~~~
+
+Suppose we want two plots to share the same color scale. This can be
+achieved by passing in the appropriate arguments and adding the color bar
+later.
+
+TODO: All xray plot methods return axes for consistency- is this necessary?
```

Comment:
According to the matplotlib folks, well behaved functions should probably return matplotlib artist objects rather than axes.

Arguably we should return whatever the main matplotlib function returns?